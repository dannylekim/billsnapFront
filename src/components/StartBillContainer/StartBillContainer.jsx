import React, {Component} from "react";
import "./styles.scss";
import StartBillButton from "../StartBillButton/StartBillButton";

class StartBillContainer extends Component {
    constructor(props) {
        super(props);
        this.handleStartBill = this.handleStartBill.bind(this);
    }

    handleStartBill(billId) {
        this.props.startABill(billId);
    }

    billCanBeStarted() {
        return this.props.userEmail === this.props.bill.responsible.email
            && this.props.bill.status === "OPEN";
    }

    render() {
        const {bill} = this.props;

        const StartActiveBill = billId => {
            return (
                <div className="button__container">
                    <StartBillButton onClickHandler={() => this.handleStartBill(billId)}/>
                </div>
            )
        }

        const displayStartBillButton = this.billCanBeStarted() ? (StartActiveBill(bill.id)) : (<div/>);

        return <>{displayStartBillButton}</>;
    }

}

export default StartBillContainer;
