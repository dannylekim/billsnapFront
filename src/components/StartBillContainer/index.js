import {startABill} from "../../redux/actions/billActions";
import {connect} from "react-redux";
import StartBillContainer from "./StartBillContainer";

const mapStateToProps = (state) => ({
  isBillLoading: state.application.isBillLoading,
  // activeBillId: state.bills.activeBill.id,
  // billResponsibleEmail: state.bills.activeBill.responsible.email,
  userEmail: state.users.userInfo.email
});

const mapDispatchToProps = (dispatch) => ({
  startABill: billId => dispatch(startABill(billId))
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(StartBillContainer);
