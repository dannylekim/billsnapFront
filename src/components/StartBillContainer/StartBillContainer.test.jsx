import React from "react";
import StartBillContainer from "./StartBillContainer";

const userEmail = "string@email.com";

const bill =
    {
        id: 123,
        name: "string 1",
        responsible: {
            firstName: "string",
            middleName: "string",
            lastName: "string",
            id: 0,
            email: "string@email.com",
        },
        status: "OPEN",
        category: "string",
        created: "2020-08-26T00:31:21.674Z",
        balance: 100,
        amountOwed: 50,
    };

describe("StartBillContainer", () => {
    describe("handleStartBill", () => {
        it("should call start bill route", async () => {
            const mockStartBill = jest.fn();
            shallow(
                <StartBillContainer startABill={mockStartBill} bill={bill} userEmail={userEmail}/>
            ).instance().handleStartBill(bill.id);

            expect(mockStartBill).toHaveBeenCalledWith(bill.id);
        });
    });

    describe("snapshots ðŸ“¸", () => {
        it("StartBillContainer should match snap shot when user is not bill responsible", () => {
            matches(<StartBillContainer bill={bill}/>);
        });

        it("StartBillContainer should match snap shot when user is bill responsible", () => {
            matches(<StartBillContainer bill={bill} userEmail={userEmail}/>);
        });
    });
});
