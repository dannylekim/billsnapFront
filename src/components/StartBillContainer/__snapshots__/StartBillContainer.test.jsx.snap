// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartBillContainer snapshots ðŸ“¸ StartBillContainer should match snap shot when user is bill responsible 1`] = `
<div
  className="button__container"
>
  <button
    className="startBill__button btn btn-primary btn-sm btn-pill"
    onClick={[Function]}
    type="button"
  >
    Start Bill
  </button>
</div>
`;

exports[`StartBillContainer snapshots ðŸ“¸ StartBillContainer should match snap shot when user is not bill responsible 1`] = `<div />`;
